core/sql/src/main/java/org/trafodion/sql/HBaseClient.java

```diff
@@ -1088,36 +1139,65 @@ public boolean estimateRowCount(String tblName, int partialRowSize,
           //printQualifiers(reader, 100);
           if (ROWS_TO_SAMPLE > 0 &&
               totalEntries == reader.getEntries()) {  // first file only
-            // Trafodion column qualifiers are ordinal numbers, which
-            // makes it easy to count missing (null) values. We also count
-            // the non-Put KVs (typically delete-row markers) to estimate
-            // their frequency in the full file set.
+
+            // Trafodion column qualifiers are ordinal numbers, but are represented
+            // as varying length unsigned little-endian integers in lexicographical
+            // order. So, for example, in a table with 260 columns, the column
+            // qualifiers (if present) will be read in this order: 
+            // 1 (x'01'), 257 (x'0101'), 2 (x'02'), 258 (x'0201'), 3 (x'03'),
+            // 259 (x'0301'), 4 (x'04'), 260 (x'0401'), 5 (x'05'), 6 (x'06'), 
+            // 7 (x'07'), ...
+            // We have crossed the boundary to the next row if and only if the
+            // next qualifier read is less than or equal to the previous, 
+            // compared unsigned, lexicographically.
```

Comment:
Some simplication is possible for aligned format though it is not clear if additional complexitu in code is warranted. With aligned format we know that each row will have exactly one column and I suppose null/default values will always be included. So the number of KeyValues we get from the header is the number of rows in the table.